lib/helpers/calculator_helpers.js

```diff
@@ -9,50 +8,42 @@ module.exports = class CalculatorHelpers {
    * Create population values (aka results) for all populations in the population set using the results from the
    * calculator.
    * @param {Measure} measure - The measure we are getting the values for.
-   * @param {Object} population - The population set that we are mapping results to.
-   * @param {Object} results - The raw results object from the calculation engine.
-   * @param {Patient} patientId - The patient ID we are getting results for.
+   * @param {Object} populationSet - The population set that we are mapping results to.
+   * @param {Object} patientResults - The raw results object from the calculation engine for a patient.
    * @param {Array} observationDefs - List of observation defines we add to the elm for calculation OBSERVs.
-   * @param {Integer} populationIndex - Index of the population set of which the population belongs to
    * @returns {[object, object]} The population results. Map of "POPNAME" to Integer result. Except for OBSERVs,
    *   their key is 'value' and value is an array of results. Second result is the the episode results keyed by
    *   episode id and with the value being a set just like the patient results.
    */
-  static createPopulationValues(measure, population, results, patientId, observationDefs, populationIndex) {
+  static createPopulationValues(measure, populationSet, patientResults, observationDefs) {
     let populationResults = {};
     let episodeResults = null;
 
     // patient based measure
-    if (!measure.episode_of_care) {
+    if (measure.calculation_method !== 'EPISODE_OF_CARE') {
       populationResults = this.createPatientPopulationValues(
-        measure,
-        population,
-        results,
-        patientId,
-        observationDefs,
-        populationIndex
+        populationSet,
+        patientResults,
+        observationDefs
       );
       populationResults = this.handlePopulationValues(populationResults);
     } else {
       // episode of care based measure
       // collect results per episode
       episodeResults = this.createEpisodePopulationValues(
-        measure,
-        population,
-        results,
-        patientId,
-        observationDefs,
-        populationIndex
+        populationSet,
+        patientResults,
+        observationDefs
       );
 
       // initialize population counts
-      Measure.ALL_POPULATION_CODES.forEach((popCode) => {
-        if (population[popCode]) {
-          if (popCode === 'OBSERV') {
-            populationResults.values = [];
-          } else {
-            populationResults[popCode] = 0;
-          }
+      Object.keys(populationSet.populations).forEach((popCode) => {
+        // Mongoose adds keys that start with '_' we do not care about these
+        if (popCode[0] !== '_') {
```

Comment:
I think we should spend some time on a different way to handle this. Ideally, they wouldn't be included in these structures.